plans/phase-1/api-design/multigres-operator-api-v1alpha1-design.md

```diff
@@ -0,0 +1,1214 @@
+      │
+      ├── 🌍 [GlobalTopoServer] (Child CR) ← 📄 Uses [CoreTemplate] OR inline [spec]
+      │
+      ├── 🤖 MultiAdmin Resources ← 📄 Uses [CoreTemplate] OR inline [spec]
+      │
+      ├── 💠 [Cell] (Child CR) ← 📄 Uses [CellTemplate] OR inline [spec]
+      │    │
+      │    ├── 🚪 MultiGateway Resources
+      │    └── 📡 [LocalTopoServer] (Child CR, optional)
+      │
+      └── 🗃️ [TableGroup] (Child CR)
+           │
+           └── 📦 [Shard] (Child CR) ← 📄 Uses [ShardTemplate] OR inline [spec]
+                │
+                ├── 🧠 MultiOrch Resources (Deployment/Pod)
+                └── 🏊 Pools (StatefulSets for Postgres+MultiPooler)
+
+📄 [CoreTemplate] (User-editable, scoped config)
+   ├── globalTopoServer
+   └── multiadmin
+
+📄 [CellTemplate] (User-editable, scoped config)
+   ├── multigateway
+   └── localTopoServer (optional)
+
+📄 [ShardTemplate] (User-editable, scoped config)
+   ├── multiorch
+   └── pools (postgres + multipooler)
+```
+
+## Design Details: API Specification
+
+### The `default` Flag (System Resources)
+
+Both `Database` and `TableGroup` entries support a boolean `default` flag (defaulting to `false`). This flag maps the definition to the **System Default** infrastructure created during the cluster's bootstrap phase.
+
+  * **On a Database (`default: true`):** Indicates this entry defines the configuration for the system-level database (typically named `postgres`) that contains the global catalog. There can be only one default database per cluster.
+  * **On a TableGroup (`default: true`):** Indicates this entry defines the configuration for the "Catch-All" or "Unsharded" group of that database. Every database has exactly one Default TableGroup where tables land by default.
+
+Defining these entries allows the user to explicitly configure the resources (replicas, storage, compute) allocated to these system components, rather than relying on hardcoded operator defaults.
+
+### User Managed CR: MultigresCluster
+
+  * This CR and the three scoped templates (`CoreTemplate`, `CellTemplate`, `ShardTemplate`) are the *only* editable entries for the end-user.
+    * All other child CRs will be owned by this top-level CR. Any manual changes to those child CRs will be immediately reverted by the `MultigresCluster` cluster controller.
+  * All component configurations (`globalTopoServer`, `multiadmin`, `cells`, `shards`) follow a consistent pattern: they can be defined via an inline `spec` or by referencing a template (`templateRef`). Providing both is a validation error.
+    * **Override Chain:** All components use the following 4-level precedence chain for configuration:
+      1.  **Component-Level Definition:** An inline `spec` or an explicit `templateRef` on the component itself.
+      2.  **Cluster-Level Default:** The corresponding template defined in `spec.templateDefaults` (e.g., `templateDefaults.coreTemplate` or `templateDefaults.cellTemplate`).
```

Comment:
I had to refer back to this section multiple times. I think this wording is VERY confusing, because I would think "cluster" would mean the Kubernetes cluster as a whole, similar to `ClusterRole` or `ClusterRoleBinding`. I understand this is about `MultigresCluster` resource, and it should be very clear that's the case.

```suggestion
      2.  **Defaults in MultigresCluster spec:** The corresponding template defined in `spec.templateDefaults` (e.g., `templateDefaults.coreTemplate` or `templateDefaults.cellTemplate`).
```